### ISSUE TITLE
Tooltip not being rendered with python.analysis.supportRestructuredText turned on.

### ISSUE THREAD

**Issue by magoocas** [microsoft/pylance-release#6284]
## Environment data

-   Pylance version: v2024.8.1
-   OS and version: win 10 22H2 build 19045.4780
-   Python version (& distribution if applicable, e.g. Anaconda): 3.12.4

## Code Snippet

```python
def add_works(a:int, b:int) -> int:
    """ Renders correctly

    Example:
    .. code-block:: python
        three = add_works(1, 2)
    """
    ...


def add_broken(a:int, b:int) -> int:
    """ Does not render

    Example:
        .. code-block:: python
            three = add_broken(1, 2)
    """
    ...

```

## Repro Steps

1. Paste code sample
2. Hover over each function to display tooltip

## Expected behavior

A tooltip is displayed for either function

## Actual behavior

A tooltip is only displayed for add_works

## Logs

2024-08-21 11:01:07.536 [info] (36844) WorkspaceFactory 0 clear
2024-08-21 11:01:07.541 [info] (Client) Experiment 'pythonaa' is active
2024-08-21 11:01:07.542 [info] (Client) Experiment 'pythonCreateEnvOnPipInstall' is active
2024-08-21 11:01:07.542 [info] (Client) Experiment 'pythonRecommendTensorboardExt' is active
2024-08-21 11:01:07.542 [info] (Client) Experiment 'pythonTerminalEnvVarActivation' is active
2024-08-21 11:01:07.542 [info] (Client) Experiment 'pythonTestAdapter' is active
2024-08-21 11:01:07.575 [info] [Info  - 11:01:07 AM] Server process exited successfully
2024-08-21 11:01:08.060 [info] (Client) Pylance async client (2024.8.1) started with python extension (2024.12.3)
2024-08-21 11:01:08.930 [info] [Info  - 11:01:08 AM] (31804) Server root directory: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist
2024-08-21 11:01:08.932 [info] [Info  - 11:01:08 AM] (31804) Experimental Pylance language server 2024.8.1 (pyright version 1.1.373, commit ee424479) starting
2024-08-21 11:01:08.939 [info] [Info  - 11:01:08 AM] (31804) Starting service instance "pylance_issue"
2024-08-21 11:01:09.008 [info] (31804) No configuration file found.
2024-08-21 11:01:09.008 [info] (31804) No pyproject.toml file found.
2024-08-21 11:01:09.009 [info] [Info  - 11:01:09 AM] (31804) Setting pythonPath for service "pylance_issue": "C:\Users\lcummin1\AppData\Local\Programs\Python\Python312\python.exe"
2024-08-21 11:01:09.010 [info] [Info  - 11:01:09 AM] (31804) Setting environmentName for service "pylance_issue": "3.12.4 (global)"
2024-08-21 11:01:09.011 [info] [Info  - 11:01:09 AM] (31804) No include entries specified; assuming c:\Users\lcummin1\repos\pylance_issue
2024-08-21 11:01:09.012 [info] [Info  - 11:01:09 AM] (31804) Auto-excluding **/node_modules
2024-08-21 11:01:09.013 [info] [Info  - 11:01:09 AM] (31804) Auto-excluding **/__pycache__
2024-08-21 11:01:09.014 [info] [Info  - 11:01:09 AM] (31804) Auto-excluding **/.*
2024-08-21 11:01:09.520 [info] [Info  - 11:01:09 AM] (31804) Assuming Python version 3.12.4.final.0
2024-08-21 11:01:09.521 [info] (31804) Assuming Python platform Windows
2024-08-21 11:01:10.279 [info] [Info  - 11:01:10 AM] (31804) Search paths for file:///c%3A/Users/lcummin1/repos/pylance_issue
2024-08-21 11:01:10.279 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\.vscode\extensions\ms-python.vscode-pylance-2024.8.1\dist\typeshed-fallback\stdlib
2024-08-21 11:01:10.280 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\repos\pylance_issue
2024-08-21 11:01:10.280 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\repos\pylance_issue\typings
2024-08-21 11:01:10.281 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\.vscode\extensions\ms-python.vscode-pylance-2024.8.1\dist\typeshed-fallback\stubs\...
2024-08-21 11:01:10.281 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\.vscode\extensions\ms-python.vscode-pylance-2024.8.1\dist\bundled\stubs
2024-08-21 11:01:10.281 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\AppData\Local\Programs\Python\Python312\DLLs
2024-08-21 11:01:10.282 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\AppData\Local\Programs\Python\Python312\Lib
2024-08-21 11:01:10.282 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\AppData\Local\Programs\Python\Python312
2024-08-21 11:01:10.283 [info] [Info  - 11:01:10 AM] (31804)   c:\Users\lcummin1\AppData\Local\Programs\Python\Python312\Lib\site-packages
2024-08-21 11:01:10.284 [info] [Info  - 11:01:10 AM] (31804) Adding fs watcher for library directories:
 file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/DLLs
file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib
file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312
file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages
2024-08-21 11:01:10.285 [info] [Info  - 11:01:10 AM] (31804) Adding fs watcher for directories:
 file:///c%3A/Users/lcummin1/repos/pylance_issue
2024-08-21 11:01:10.286 [info] (31804) Searching for source files
2024-08-21 11:01:10.293 [info] [Info  - 11:01:10 AM] (31804) Found 1 source file
2024-08-21 11:01:10.357 [info] (31804) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2024-08-21 11:01:10.639 [info] (31804) Attempting to resolve using local imports: __builtins__
2024-08-21 11:01:10.644 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (279ms)
2024-08-21 11:01:10.855 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 84ms] (210ms)
2024-08-21 11:01:10.937 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/builtins.pyi (79ms)
2024-08-21 11:01:10.938 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (1ms)
2024-08-21 11:01:10.944 [info] [Info  - 11:01:10 AM] (31804) Background analysis(1) root directory: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist
2024-08-21 11:01:10.946 [info] [Info  - 11:01:10 AM] (31804) Background analysis(1) started
2024-08-21 11:01:10.947 [info] (31804) pytest configurations: {"message":"request cancelled","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2024-08-21 11:01:10.949 [info] (31804) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2024-08-21 11:01:11.022 [info] (31804) pytest configurations: {"message":"script","classes":["Test"],"files":["test_*.py","*_test.py"],"functions":["test"]}
2024-08-21 11:01:11.230 [info] [Info  - 11:01:11 AM] (31804) Indexer background runner(2) root directory: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist (index)
2024-08-21 11:01:11.231 [info] [Info  - 11:01:11 AM] (31804) Indexing(2) started
2024-08-21 11:01:11.537 [info] (31804) Attempting to resolve using local imports: __builtins__
2024-08-21 11:01:11.539 [info] (31804) [BG(1)] analyzing: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py ...
2024-08-21 11:01:11.539 [info] (31804) [BG(1)]   parsing: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (289ms)
2024-08-21 11:01:11.656 [info] (31804) [BG(1)]   parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/builtins.pyi [fs read 4ms] (115ms)
2024-08-21 11:01:11.713 [info] (31804) [BG(1)]   binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/builtins.pyi (54ms)
2024-08-21 11:01:11.714 [info] (31804) [BG(1)]   binding: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (0ms)
2024-08-21 11:01:11.784 [info] (31804) [BG(1)]   checking: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py ...
2024-08-21 11:01:11.784 [info] (31804) [BG(1)]     parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 7ms] (33ms)
2024-08-21 11:01:11.807 [info] (31804) [BG(1)]     binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/typing.pyi (21ms)
2024-08-21 11:01:11.840 [info] (31804) [BG(1)]     parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 2ms] (13ms)
2024-08-21 11:01:11.848 [info] (31804) [BG(1)]     binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/types.pyi (8ms)
2024-08-21 11:01:11.866 [info] (31804) [BG(1)]     parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 2ms] (8ms)
2024-08-21 11:01:11.885 [info] (31804) [BG(1)]     binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (16ms)
2024-08-21 11:01:11.899 [info] (31804) [BG(1)]     parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 2ms] (4ms)
2024-08-21 11:01:11.901 [info] (31804) [BG(1)]     binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/abc.pyi (2ms)
2024-08-21 11:01:11.938 [info] (31804) [BG(1)]   checking: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (220ms)
2024-08-21 11:01:11.939 [info] [Info  - 11:01:11 AM] (31804) Heap stats: total_memory_size=32684MB, total_free_size=7119MB, total_heap_size=53MB, used_heap_size=37MB, cross_worker_used_heap_size=37MB, total_physical_size=53MB, total_available_size=4057MB, heap_size_limit=4096MB
2024-08-21 11:01:11.939 [info] (31804) [BG(1)] analyzing: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py (685ms)
2024-08-21 11:01:11.967 [info] (31804) [BG(1)] indexing: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py [found 2] (0ms)
2024-08-21 11:01:11.968 [info] (31804) Indexing Done: file:///c%3A/Users/lcummin1/repos/pylance_issue/example.py
2024-08-21 11:01:12.110 [info] (31804) [IDX(2)] scan packages file:///c%3A/Users/lcummin1/repos/pylance_issue ...
2024-08-21 11:01:12.110 [info] (31804) [IDX(2)]   read stdlib indices (19ms)
2024-08-21 11:01:13.382 [info] (31804) [IDX(2)] scan packages file:///c%3A/Users/lcummin1/repos/pylance_issue (1291ms)
2024-08-21 11:01:13.383 [info] [Info  - 11:01:13 AM] (31804) scanned(2) 60 files over 1 exec env
2024-08-21 11:01:13.739 [info] (31804) [IDX(2)] index packages file:///c%3A/Users/lcummin1/repos/pylance_issue ...
2024-08-21 11:01:13.739 [info] (31804) [IDX(2)]   index execution environment file:///c%3A/Users/lcummin1/repos/pylance_issue ...
2024-08-21 11:01:13.740 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/concurrent/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.740 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/idlelib/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.741 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/lib2to3/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.742 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/pydoc_data/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.743 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/test/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.744 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/turtledemo/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.745 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/urllib/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.745 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/wsgiref/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.746 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/xmlrpc/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.747 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/__phello__/__init__.py [skipped: no '__all__' defined] (0ms)
2024-08-21 11:01:13.748 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/binaryornot/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.749 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/cookiecutter/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.750 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/distlib/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.751 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/msgpack/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.752 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/poetry_plugin_export/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.753 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/ptyprocess/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.754 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/shellingham/__init__.py [skipped: no '__all__' defined] (0ms)
2024-08-21 11:01:13.755 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/text_unidecode/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.756 [info] (31804) [IDX(2)]     indexing: file:///c%3A/Users/lcummin1/AppData/Local/Programs/Python/Python312/Lib/site-packages/win32ctypes/__init__.py [skipped: no '__all__' defined] (1ms)
2024-08-21 11:01:13.762 [info] (31804) [IDX(2)]   index execution environment file:///c%3A/Users/lcummin1/repos/pylance_issue [found 1438 in 41 files] (31ms)
2024-08-21 11:01:13.763 [info] (31804) [IDX(2)] index packages file:///c%3A/Users/lcummin1/repos/pylance_issue [found 1438 in 1 exec envs] (95ms)
2024-08-21 11:01:13.763 [info] [Info  - 11:01:13 AM] (31804) indexed(2) 41 files over 1 exec env
2024-08-21 11:01:13.820 [info] [Info  - 11:01:13 AM] (31804) Indexing finished(2).
2024-08-21 11:01:16.482 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/typing.pyi [fs read 7ms] (51ms)
2024-08-21 11:01:16.497 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/typing.pyi (12ms)
2024-08-21 11:01:16.527 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/types.pyi [fs read 2ms] (16ms)
2024-08-21 11:01:16.537 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/types.pyi (9ms)
2024-08-21 11:01:16.556 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi [fs read 1ms] (9ms)
2024-08-21 11:01:16.560 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/_typeshed/__init__.pyi (5ms)
2024-08-21 11:01:16.573 [info] (31804) [FG] parsing: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/abc.pyi [fs read 2ms] (3ms)
2024-08-21 11:01:16.574 [info] (31804) [FG] binding: file:///c%3A/Users/lcummin1/.vscode/extensions/ms-python.vscode-pylance-2024.8.1/dist/typeshed-fallback/stdlib/abc.pyi (1ms)
2024-08-21 11:01:16.578 [info] (31804) Restructured text parsing enabled
2024-08-21 11:01:16.579 [info] (31804) Converting docstring to markdown
2024-08-21 11:01:16.579 [info] (31804)  Renders correctly

    Example:
    .. code-block:: python
        three = add_works(1, 2)
    
2024-08-21 11:01:16.620 [info] (31804) Fixed docstring:
2024-08-21 11:01:16.621 [info] (31804)  Renders correctly

Examples
--------

----

.. code-block:: python
    three = add_works(1, 2)
    

2024-08-21 11:01:16.624 [info] (31804) Conversion complete in 45ms
2024-08-21 11:01:16.625 [info] (31804) 
Renders correctly

## Examples

``` python
    three = add_works(1, 2)
```

2024-08-21 11:01:18.692 [info] (31804) Restructured text parsing enabled
2024-08-21 11:01:18.693 [info] (31804) Converting docstring to markdown
2024-08-21 11:01:18.694 [info] (31804)  Does not render

    Example:
        .. code-block:: python
            three = add_broken(1, 2)
    
2024-08-21 11:01:18.696 [info] (31804) Fixed docstring:
2024-08-21 11:01:18.697 [info] (31804)  Does not render

Examples
--------
.. code-block:: python
three = add_broken(1, 2)
    
2024-08-21 11:01:18.699 [info] [Error - 11:01:18 AM] Request textDocument/hover failed.
2024-08-21 11:01:18.699 [info]   Message: Request textDocument/hover failed with message: Cannot read properties of undefined (reading 'type')
  Code: -32603 

```
XXX
```

**Comment by rchiodo** [comment]
Thanks I can reproduce. Should be able to fix this for our next release.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2024.8.103, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20248103-27-August-2024-PreRelease)
